src/main/scala/nl.knaw.dans.easy.bag2deposit/FilesXml.scala

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (C) 2020 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.easy.bag2deposit
+
+import nl.knaw.dans.lib.logging.DebugEnhancedLogging
+
+import scala.xml.transform.{RewriteRule, RuleTransformer}
+import scala.xml.{Elem, NamespaceBinding, Node, NodeSeq, XML}
+
+object FilesXml extends DebugEnhancedLogging {
+
+  val DEFAULT_PREFIX = "dcterms"
+  val DEFAULT_URI = "http://purl.org/dc/terms/"
+
+  def apply(filesXml: Elem, destination: String, addedFiles: Seq[String], mimeType: String): Node = {
+
+    val formatTagPrefix = Option(filesXml.scope.getPrefix(DEFAULT_URI)).getOrElse(DEFAULT_PREFIX)
+    val filesXmlWithPossiblyAddedNamespace  = Option(filesXml.scope.getURI(formatTagPrefix))
+      .map(_ => filesXml)
+      .getOrElse(filesXml.copy(scope = NamespaceBinding(formatTagPrefix, DEFAULT_URI, filesXml.scope)))
+    val newFileElements = addedFiles.foldLeft(NodeSeq.Empty)((a,b) => a ++
+      XML.loadString(
+      s"""<file filepath="$destination/$b">
+           <$formatTagPrefix:format>$mimeType</$formatTagPrefix:format>
+        </file>"""))
```

Comment:
Don't parse string but use embedded syntax as below in the rewrite rule or for example https://github.com/DANS-KNAW/easy-convert-bag-to-deposit/blob/57e4ab9513d536c16121ad8916058d4102154138/src/main/scala/nl.knaw.dans.easy.bag2deposit/ddm/AcquisitionRewriteRule.scala#L53-L57

Reply:
The problem here is that also the `tag prefix` is variable, that is why I use `XML.loadString`

Reply:
line 46 shows how to solve that.

Reply:
have you tried it?